src/wasm/wasm.cpp

```diff
@@ -940,7 +940,9 @@ size_t Function::getNumLocals() { return params.size() + vars.size(); }
 
 bool Function::isParam(Index index) { return index < params.size(); }
 
-bool Function::isVar(Index index) { return index >= params.size(); }
+bool Function::isVar(Index index) {
+  return params.size() <= index && index < params.size() + vars.size();
```

Comment:
I think it would be even better to assert that the index is in bounds, as the assumption in this API is "this is a valid local index, is it a var?" But we can leave that for separate consideration.